lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -2919,39 +3403,22 @@ int s1ap_mme_handle_erab_rel_response(
   S1AP_E_RAB_REL_RSP(message_p).e_rab_rel_list.no_of_items           = 1;
   S1AP_E_RAB_REL_RSP(message_p).e_rab_failed_to_rel_list.no_of_items = 0;
 
-  if (s1ap_E_RABReleaseResponseIEs_p->presenceMask &
-      S1AP_E_RABRELEASERESPONSEIES_E_RABRELEASELISTBEARERRELCOMP_PRESENT) {
-    int num_erab = s1ap_E_RABReleaseResponseIEs_p->e_RABReleaseListBearerRelComp
-                       .s1ap_E_RABReleaseItemBearerRelComp.count;
+  const S1ap_E_RABList_t* const e_rab_list = &ie->value.choice.E_RABList;
+  int num_erab                             = e_rab_list->list.count;
+  if (ie) {
     for (int index = 0; index < num_erab; index++) {
-      S1ap_E_RABReleaseItemBearerRelComp_t* erab_rel_item =
-          (S1ap_E_RABReleaseItemBearerRelComp_t*)
-              s1ap_E_RABReleaseResponseIEs_p->e_RABReleaseListBearerRelComp
-                  .s1ap_E_RABReleaseItemBearerRelComp.array[index];
+      const S1ap_E_RABItemIEs_t* const erab_item_ies =
+          (S1ap_E_RABItemIEs_t*) e_rab_list->list.array[index];
+      const S1ap_E_RABItem_t* const erab_item =
+          (S1ap_E_RABItem_t*) &erab_item_ies->value.choice.E_RABItem;
       S1AP_E_RAB_REL_RSP(message_p).e_rab_rel_list.item[index].e_rab_id =
-          erab_rel_item->e_RAB_ID;
-      S1AP_E_RAB_REL_RSP(message_p).e_rab_rel_list.no_of_items += 1;
-    }
-  }
-
-  if (s1ap_E_RABReleaseResponseIEs_p->presenceMask &
```

Comment:
We also need to check for ERab_failed_to_release list IE in the erab_release_response IEs.

Reply:
Resolved